sycl/source/detail/memory_manager.cpp

```diff
@@ -187,16 +187,31 @@ void copyH2D(SYCLMemObjT *SYCLMemObj, char *SrcMem, QueueImplPtr SrcQueue,
         /*blocking_write=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
         SrcMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
   } else {
-    size_t BufferRowPitch = (1 == DimSrc) ? 0 : SrcSize[0];
-    size_t BufferSlicePitch = (3 == DimSrc) ? SrcSize[0] * SrcSize[1] : 0;
 
-    size_t HostRowPitch = (1 == DimDst) ? 0 : DstSize[0];
-    size_t HostSlicePitch = (3 == DimDst) ? DstSize[0] * DstSize[1] : 0;
-    CHECK_OCL_CODE(clEnqueueWriteBufferRect(
-        CLQueue, DstMem,
-        /*blocking_write=*/CL_FALSE, &DstOffset[0], &SrcOffset[0],
-        &DstAccessRange[0], BufferRowPitch, BufferSlicePitch, HostRowPitch,
-        HostSlicePitch, SrcMem, DepEvents.size(), &DepEvents[0], &OutEvent));
+    if( SrcSize[0]   == SrcAccessRange[0] &&  SrcSize[1]   == SrcAccessRange[1]  &&  SrcSize[2]   == SrcAccessRange[2] &&
+	SrcSize[0]   == DstSize[0]        &&  SrcSize[1]   == DstSize[1]         &&  SrcSize[2]   == DstSize[2]        &&
+	SrcSize[0]   == DstAccessRange[0] &&  SrcSize[1]   == DstAccessRange[1]  &&  SrcSize[2]   == DstAccessRange[2] &&
+	SrcOffset[0] == DstOffset[0]      &&  SrcOffset[1] == DstOffset[1]       &&  SrcOffset[2] == DstOffset[2]      &&
+	SrcOffset[0] == 0                 &&  SrcOffset[1] == 0                  &&  SrcOffset[2] == 0 ) {
+      
+      CHECK_OCL_CODE(clEnqueueWriteBuffer(
+	  CLQueue, DstMem,
+	  /*blocking_write=*/CL_FALSE, DstOffset[0], DstAccessRange[0]*DstAccessRange[1]*DstAccessRange[2],
+	  SrcMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
+      
+    } else {
+    
```

Comment:
No need for the excess white space, especially as it's not consistent across your changes